drivers/hmis/app/views/hmis_client/services/_table.haml

```diff
@@ -0,0 +1,31 @@
+.table-responsive
+  %table.table
+    %thead
+      %tr
+        %th
+        %th Project
+        %th Entry Date
+        %th Date of Service
+        %th Service Type
+        %th
+    %tbody
+      - custom_services.each do |service|
+        %tr
+          %td
+            .ds{class: "ds-color-#{service.data_source_id}"}= ds_short_name_for(service.client.id)
+          %td
+            = cache(service.project.as_warehouse) do
+              = service.project.as_warehouse.name(current_user)
```

Comment:
The thing we want to cache is the `as_warehouse` lookup, so that lookup should be in the yielded block, not the cache key. I think that defeats the purpose.   https://apidock.com/rails/ActionView/Helpers/CacheHelper/cache

@eanders what do you recommend for railsy way to do the cache key?

maybe something like `[service, current_user, can_view_confidential_project_names?], expires_in: 4.hours` ?

Reply:
Hmm, I'm not really a fan of the partial caching, I wonder if we can just add memost to the `as_warehouse` method.  Should be fine to do the lookup once per page per project right?  I think the goal is to not look up the `GrdaWarehouse::Hud::Project` record for every service.

I need to think on this one a bit more.

Reply:
2 other ideas:
* add an extension for warehouse_project, and have the rollup method `preload(:warehouse_project)`. and this call `service.warehouse_project.name(user)`. Extensions are at `drivers/hmis/extensions/grda_warehouse/hud`
* move confidential name logic to a shared concern, so we can call `service.project.confidentialized_warehouse_name(user)` or whatnot

Reply:
I like option 1, but option 2 would be fine with me as well.  Thanks @gigxz !

Reply:
Thank you both! @gigxz could you expand on the reason to put it in an extension rather than just putting it on the model? Also, since the model we are extending is (I think) `Hmis::Hud::CustomService`, would the extension go in `drivers/hmis_external_apis/extensions/hmis/hud/` (rather than `drivers/hmis/extensions/grda_warehouse/hud`)?

Reply:
@martha, pretty sure that should be a relation on `GrdaWarehouse::Hud::Project`.  Though maybe it would be cleaner on service, in which case, yeah, I think it would just go on the model.

Reply:
Ah so, if it should be an extension on Project rather than CustomService, then instead of `service.warehouse_project.name(user)` as Gig wrote above, it would be `service.project.warehouse_project.name(user)`?

Reply:
Okay! Makes sense, thank you.